### ISSUE TITLE
errors when trying to work with UnrealCV plugin: "Plugin missing or incompatible"

### ISSUE THREAD

**Issue by mikeroberts3000** [unrealcv/unrealcv#25]
Hello there,

I'm attempting to install the UnrealCV plugin into the Unreal Engine. I'm using Unreal Engine 4.12.5 on OSX. I've followed the instructions on this page:

https://unrealcv.github.io/tutorial/plugin.html

But when I try to do the "Open Menu -> Edit -> Plugins, make sure UnrealCV is installed and enabled" step, I get the following error when I try to load an Unreal project. For what it's worth, I'm attempting to load the EpicZenGarden uproject from the Epic Games Launcher app, after I've installed the UnrealCV plugin, both in the project Plugins folder and in the Engine/Plugins folder, following the UnrealCV tutorial instructions.

![image](https://cloud.githubusercontent.com/assets/2341965/19018925/db266592-8827-11e6-9087-1ca58a974cd5.png)

If I click "No", then I get more errors:

![image](https://cloud.githubusercontent.com/assets/2341965/19019018/44294520-882b-11e6-8f91-96b9c36cd195.png)

When I click "OK", I get another error:

![image](https://cloud.githubusercontent.com/assets/2341965/19019023/5c3be636-882b-11e6-83c5-b7c6ca5fcb9f.png)

I realize I'm on charting new ground here, trying to get this running on OSX. Is there an easy hack I can do to work around these errors?  :)

**Comment by qiuwch** [comment]
Hi, @mikeroberts3000:

Sorry for this inconvenience. I did not include a compiled version of the plugin for Mac yet.

To solve this, there are two solutions.
1. You can create a C++ unreal project, when you open that project. The unrealcv will be compiled. If it is just a blueprint project, no compilation will be done. 
2. I can provide you a compiled version for mac. I can send you with email tonight. I did not have my mac with me right now.

**Comment by mikeroberts3000** [comment]
@qiuwch Do you mean a compiled version of the UnrealCV plugin? Interestingly, the content downloaded by running `curl -L http://unrealcv.github.io/install.sh | sh` (as specified in the tutorial) doesn't contain any binaries. But then after you try to open the project in Unreal, a `Binaries` folder gets created with `Win64` and `Linux` subfolders, but no `Mac64` folder. Is this what you mean? Is there a special flag you need to set somewhere to make Unreal compile the right binaries for Mac?

Of the solutions you suggested, I think solution 2 is better for the UnrealCV project. Because then there is one recommended way to do everything, only one tutorial to write, etc.

**Comment by qiuwch** [comment]
@mikeroberts3000 

That is a bit strange. The downloaded zip file should already contain `Binaries/Win64`, `Binaries/Linux` without needing to run Unreal. I just tried this.

I totally agree with you. The missing binary for mac is something I need to do. There are also many unfinished parts of the website I need to spend time with. Currently, the building process is: I need to build the plugin in three different machines with different OS, then merge the compiled files together to create a zip file.

I spent about a week to setting up the Jenkins CI on three machines trying to automate this for the future releases. Let's see whether it works well.
